### PR TITLE
grub installation: integrate trustedGRUB + fix broken equality check

### PR DESCRIPTION
**Pull request by ts468**
Depends on PR #8647.

### CODE REVIEW

**Comment by ts468** [comment]
@wkennington, could you maybe have a look at the changes in install-grub.pl, please? There seemed to be an error in the equality check system within install-grub.pl that should be fixed with this PR.

**Comment by ts468** [comment]
@wkennington, thanks a lot! Could you maybe also merge PR #8647, please?

File: nixos/modules/system/boot/loader/grub/install-grub.pl [hunk 1]
```diff
@@ -497,10 +500,11 @@ sub getEfiTarget {
 my @prevDeviceTargets = split/:/, $prevGrubState->devices;
 
 my $devicesDiffer = scalar (List::Compare->new( '-u', '-a', \@deviceTargets, \@prevDeviceTargets)->get_symmetric_difference() );
-my $versionDiffer = (get("fullVersion") eq \$prevGrubState->version);
-my $efiDiffer = ($efiTarget eq \$prevGrubState->efi);
-my $efiMountPointDiffer = ($efiSysMountPoint eq \$prevGrubState->efiMountPoint);
-my $requireNewInstall = $devicesDiffer || $versionDiffer || $efiDiffer || $efiMountPointDiffer || (($ENV{'NIXOS_INSTALL_GRUB'} // "") eq "1");
+my $nameDiffer = !(get("fullName") eq $prevGrubState->name);
```
Comment (edolstra) [inline nixos/modules/system/boot/loader/grub/install-grub.pl]: Why `!(... eq ...)` instead of `... ne ...`?

**Comment by edolstra** [comment]
What is trusted GRUB?

File: nixos/modules/system/boot/loader/grub/install-grub.pl (hunk 1 above)
Comment (ts468) [inline nixos/modules/system/boot/loader/grub/install-grub.pl]: Because my Perl skills are rather limited ;)

**Comment by ts468** [comment]
TrustedGRUB is a TPM enabled GRUB2 Bootloader (https://github.com/Sirrix-AG/TrustedGRUB2).

File: nixos/modules/system/boot/loader/grub/install-grub.pl (hunk 1 above)
Comment (ts468) [inline nixos/modules/system/boot/loader/grub/install-grub.pl]: @edolstra, thanks! Here the change: #8655.

**Pull request merged**